beetmoverscript/script.py

```diff
@@ -31,12 +32,15 @@
     context.task = get_task(context.config)  # e.g. $cfg['work_dir']/task.json
     # 2. validate the task
     validate_task_schema(context)
-    # 3 grab manifest props with all the useful data
+    # 3 prepare manifest props file
+    #   a. grab manifest props with all the useful data
+    #   b. amend its platform field to proper one
     context.properties = await get_props(context)
+    amend_props(context, PLATFORM_MAP)
```

Comment:
Really liked this comment and suggestion 👍 I wasn't pretty happy about my workaround either but somehow I moved forward. But "no-side effects" is a good desideratum so I'll gladly comply ;)